pyproject.toml

```diff
@@ -0,0 +1,207 @@
+
+[project.scripts]
+superset = "superset.cli.main:superset"
+
+[project.entry-points."sqlalchemy.dialects"]
+superset = "superset.extensions.metadb:SupersetAPSWDialect"
+postgres = "sqlalchemy.dialects.postgresql:dialect"
+"postgres.psycopg2" = "sqlalchemy.dialects.postgresql:dialect"
+
+[project.entry-points."shillelagh.adapter"]
+superset = "superset.extensions.metadb:SupersetShillelaghAdapter"
+
+[project.optional-dependencies]
+athena = ["pyathena[pandas]>=2, <3"]
+aurora-data-api = ["preset-sqlalchemy-aurora-data-api>=0.2.8,<0.3"]
+bigquery = [
+    "pandas-gbq>=0.19.1",
+    "sqlalchemy-bigquery>=1.6.1",
+    "google-cloud-bigquery>=3.10.0",
+]
+clickhouse = ["clickhouse-connect>=0.5.14, <1.0"]
+cockroachdb = ["cockroachdb>=0.3.5, <0.4"]
+cors = ["flask-cors>=2.0.0"]
+crate = ["crate[sqlalchemy]>=0.26.0, <0.27"]
+databend = ["databend-sqlalchemy>=0.3.2, <1.0"]
+databricks = [
+    "databricks-sql-connector>=2.0.2, <3",
+    "sqlalchemy-databricks>=0.2.0",
+]
+db2 = ["ibm-db-sa>0.3.8, <=0.4.0"]
+doris = ["pydoris>=1.0.0, <2.0.0"]
+dremio = ["sqlalchemy-dremio>=1.1.5, <1.3"]
+drill = ["sqlalchemy-drill>=1.1.4, <2"]
+druid = ["pydruid>=0.6.5,<0.7"]
+duckdb = ["duckdb-engine>=0.9.5, <0.10"]
+dynamodb = ["pydynamodb>=0.4.2"]
+solr = ["sqlalchemy-solr >= 0.2.0"]
+elasticsearch = ["elasticsearch-dbapi>=0.2.9, <0.3.0"]
+exasol = ["sqlalchemy-exasol >= 2.4.0, <3.0"]
+excel = ["xlrd>=1.2.0, <1.3"]
+firebird = ["sqlalchemy-firebird>=0.7.0, <0.8"]
+firebolt = ["firebolt-sqlalchemy>=1.0.0, <2"]
+gevent = ["gevent>=23.9.1"]
+gsheets = ["shillelagh[gsheetsapi]>=1.2.10, <2"]
+hana = ["hdbcli==2.4.162", "sqlalchemy_hana==0.4.0"]
+hive = [
+    "pyhive[hive]>=0.6.5;python_version<'3.11'",
+    "pyhive[hive_pure_sasl]>=0.7.0",
+    "tableschema",
+    "thrift>=0.14.1, <1.0.0",
+]
+impala = ["impyla>0.16.2, <0.17"]
+kusto = ["sqlalchemy-kusto>=2.0.0, <3"]
+kylin = ["kylinpy>=2.8.1, <2.9"]
+mssql = ["pymssql>=2.2.8, <3"]
+mysql = ["mysqlclient>=2.1.0, <3"]
+netezza = ["nzalchemy>=11.0.2"]
+ocient = [
+    "sqlalchemy-ocient>=1.0.0",
+    "pyocient>=1.0.15, <2",
+    "shapely",
+    "geojson",
+]
+oracle = ["cx-Oracle>8.0.0, <8.1"]
+pinot = ["pinotdb>=0.3.3, <0.4"]
+playwright = ["playwright>=1.37.0, <2"]
+postgres = ["psycopg2-binary==2.9.6"]
+presto = ["pyhive[presto]>=0.6.5"]
+trino = ["trino>=0.328.0"]
+prophet = ["prophet>=1.1.5, <2"]
+redshift = ["sqlalchemy-redshift>=0.8.1, <0.9"]
+rockset = ["rockset-sqlalchemy>=0.0.1, <1"]
+shillelagh = ["shillelagh[all]>=1.2.10, <2"]
+snowflake = ["snowflake-sqlalchemy>=1.2.4, <2"]
+spark = [
+    "pyhive[hive]>=0.6.5;python_version<'3.11'",
+    "pyhive[hive_pure_sasl]>=0.7",
+    "tableschema",
+    "thrift>=0.14.1, <1",
+]
+starrocks = ["starrocks>=1.0.0"]
+teradata = ["teradatasql>=16.20.0.23"]
+thumbnails = ["Pillow>=10.0.1, <11"]
+vertica = ["sqlalchemy-vertica-python>=0.5.9, < 0.6"]
+development = [
+	"docker",
+	"flask-testing",
+	"freezegun",
+	"greenlet>=2.0.2",
```

Comment:
Isn't `greenlet` a requirement for `gevent`?

Reply:
yup, as `requirements/development.txt` compile, `greenlet` gets imported via all of these -> 
```
greenlet==3.0.3
    # via
    #   apache-superset (setup.py)
    #   gevent
    #   playwright
    #   shillelagh
```

the reason why it made this `development` group is that somehow it was listed here https://github.com/apache/superset/blob/master/requirements/docker.in#L19 , so I merged that under development

Reply:
Seems related to https://github.com/apache/superset/issues/26396